elements/pl-hidden-hints/pl-hidden-hints.py

```diff
@@ -0,0 +1,80 @@
+import chevron
+import lxml.html
+import prairielearn as pl
+from typing import TypedDict, Tuple, List
+
+
+class HintsDict(TypedDict):
+    """A class with type signatures for the partial scores dict"""
+    html_hints: List[Tuple[int, str]]
+
+
+def get_param_key(name_prefix: str) -> str:
+    return f'{name_prefix}_hidden_hints'
+
+
+def prepare(element_html: str, data: pl.QuestionData) -> None:
+    element = lxml.html.fragment_fromstring(element_html)
+    pl.check_attribs(element, [], ['name'])
+
+    name_prefix = pl.get_string_attrib(element, 'name', '')
+    param_key = get_param_key(name_prefix)
+
+    # Set param dict using prefix. The use of a prefix allows for multiple hint panels.
+    if (param_key in data['params']):
+        raise ValueError(f"Duplicate params key: '{param_key}'.")
+
+    hints_dict = data['params'].setdefault(param_key, dict())
+
+    # Parse hints from frontend
+    html_hints = []
+
+    # Read hints defined as child tags and load them into html_hints
+    for i, child in enumerate(element):
+        if child.tag == 'pl-hint':
+            pl.check_attribs(child, [], ['show-after'])
+
+            # Default show-after to -1 to automatically show hint (closed) at start
+            priority = pl.get_integer_attrib(child, 'show-after', -1)
+            html_hints.append((priority, pl.inner_html(child)))
+        else:
+            raise Exception(f"Child tag type '{child.tag}' not supported by pl-hidden-hints.")
```

Comment:
Very minor: the error message would be a little more useful if it told the user what to do, instead of just telling them what not to do. Maybe "Tags inside of `pl-hiddent-hints` must be `pl-hint`. 

This is not required, but you could also maybe allow HTML comment nodes, and just do nothing with them. We do this with `pl-order-blocks` so people can leave themself comments in their HTML if they want.